pepper-apis/src/main/java/org/broadinstitute/ddp/service/AddressService.java

```diff
@@ -281,6 +282,23 @@ public boolean deleteAddress(Handle handle, String addressGuid) {
         return findDefaultAddressForParticipant(handle, participantGuid, DEFAULT_OLC_PRECISION);
     }
 
+    public Optional<MailAddress> findDefaultAddressForParticipantWithRetries(Handle handle, String participantGuid, int retries) {
+        do {
+            Optional<MailAddress> optionalAddress = findDefaultAddressForParticipant(handle, participantGuid);
+            if (optionalAddress.isPresent()) {
+                return optionalAddress;
+            } else {
+                try {
+                    LOG.warn("Could not find default address for participant with guid {}. Waiting to retry", participantGuid);
+                    TimeUnit.SECONDS.sleep(2);
+                } catch (InterruptedException e) {
+                    LOG.error("Sleep while waiting to retry findDefaultAddressForParticipant was interrupted, e");
```

Comment:
logic lgtm, that amount of retries looks like it'll do the trick - one comment - I think the log.error() in the catch might have the ending quotation marks in the wrong place since it's at the end encasing the error message.

Reply:
Thnx! Fixed here cf5a4444bc6d7cc8a7edda78fec3f8e326951d1e
I need have you review more of my PRs, @Rinchan20  !